server/internal/server/user.go

```diff
@@ -0,0 +1,78 @@
+package server
+
+import (
+	"context"
+
+	gerrors "github.com/llmariner/common/pkg/gormlib/errors"
+	"github.com/llmariner/common/pkg/id"
+	v1 "github.com/llmariner/user-manager/api/v1"
+	"github.com/llmariner/user-manager/pkg/userid"
+	"github.com/llmariner/user-manager/server/internal/store"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// CreateInternalUser creates a new user and related organization and project, and generates an API key.
```

Comment:
Could you add a TDDO comment on running all DB operations in a single transaction?